src/lib/utils/meta_tables/meta_system_utilization_table.cpp

```diff
@@ -0,0 +1,353 @@
+
+  Fail("Method not implemented for this platform");
+}
+
+float MetaSystemUtilizationTable::_get_system_cpu_usage() {
+#if defined __linux__
+
+  static uint64_t last_user_time = 0u, last_user_nice_time = 0u, last_kernel_time = 0u, last_idle_time = 0u;
+
+  std::ifstream stat_file;
+  stat_file.open("/proc/stat", std::ifstream::in);
+
+  std::string cpu_line;
+  std::getline(stat_file, cpu_line);
+  uint64_t user_time, user_nice_time, kernel_time, idle_time;
+  std::sscanf(cpu_line.c_str(), "cpu %lu %lu %lu %lu", &user_time, &user_nice_time, &kernel_time, &idle_time);
+  stat_file.close();
+
+  auto used = (user_time - last_user_time) + (user_nice_time - last_user_nice_time) + (kernel_time - last_kernel_time);
+  auto total = used + (idle_time - last_idle_time);
+
+  last_user_time = user_time;
+  last_user_nice_time = user_nice_time;
+  last_kernel_time = kernel_time;
+  last_idle_time = idle_time;
+
+  auto cpus = _get_cpu_count();
+
+  return (100.0 * used) / (total * cpus);
+
+#elif defined __APPLE__
+
+  static uint64_t last_total_ticks = 0u, last_idle_ticks = 0u;
+
+  host_cpu_load_info_data_t cpu_info;
+  mach_msg_type_number_t count = HOST_CPU_LOAD_INFO_COUNT;
+  if (host_statistics(mach_host_self(), HOST_CPU_LOAD_INFO, reinterpret_cast<host_info_t>(&cpu_info), &count) !=
+      KERN_SUCCESS) {
+    Fail("Unable to access host_statistics");
+  }
+
+  uint64_t total_ticks = 0;
+  for (int cpu_state = 0; cpu_state <= CPU_STATE_MAX; ++cpu_state) {
+    total_ticks += cpu_info.cpu_ticks[cpu_state];
+  }
+  auto idle_ticks = cpu_info.cpu_ticks[CPU_STATE_IDLE];
+
+  auto total = total_ticks - last_total_ticks;
+  auto idle = idle_ticks - last_idle_ticks;
+
+  last_total_ticks = total_ticks;
+  last_idle_ticks = idle_ticks;
+
+  auto cpus = _get_cpu_count();
+
+  return 100.0f * (1.0f - (static_cast<float>(idle) / static_cast<float>(total))) / cpus;
+
+#endif
+
+  Fail("Method not implemented for this platform");
+}
+
+float MetaSystemUtilizationTable::_get_process_cpu_usage() {
+#if defined __linux__
+
+  static clock_t last_clock_time = 0u, last_kernel_time = 0u, last_user_time = 0u;
+  struct tms timeSample;
+
+  auto clock_time = times(&timeSample);
+  auto kernel_time = timeSample.tms_stime;
+  auto user_time = timeSample.tms_utime;
+
+  auto used = (user_time - last_user_time) + (kernel_time - last_kernel_time);
+  auto total = clock_time - last_clock_time;
+
+  last_user_time = user_time;
+  last_kernel_time = kernel_time;
+  last_clock_time = clock_time;
+
+  auto cpus = _get_cpu_count();
+
+  return (100.0 * used) / (total * cpus);
+
+#elif defined __APPLE__
+
+  static uint64_t last_clock_time = 0u, last_system_time = 0u, last_user_time = 0u;
+
+  mach_timebase_info_data_t info;
+  mach_timebase_info(&info);
+
+  uint64_t clock_time = mach_absolute_time();
+
+  struct rusage resource_usage;
+  if (getrusage(RUSAGE_SELF, &resource_usage)) {
+    Fail("Unable to access rusage");
+  }
+
+  uint32_t nano = 1'000'000'000, micro = 1'000;

```

Comment:
These should be constants which should also be indicated by writing their name in all caps. Even better, we should be able to use these, shouldn't we? https://de.cppreference.com/w/cpp/numeric/ratio/ratio